### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by techmoves**
In this pull request the following tasks were implemented :

Each class from the [UML class diagram](https://github.com/Jolak5/Catalog-Ruby/pull/images/catalog_of_my_things.png) should be created in a separate .rb file.
    All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
    All properties visible in the diagram should be defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.
 All methods visible in the diagram should be implemented:
        add_item method (in all classes that have that method)
            should take an instance of the Item class as an input.
            should add the input item to the collection of items.
            should add self as a property of the item object (by using the correct setter from the item object).
        can_be_archived?() in the Item class
            should return true if published_date is older than 10 years.
            otherwise, it should return false.
        move_to_archive() in the Item class
            should reuse can_be_archived?() method.
            should change the archived property to true if the result of the can_be_archived?() method is true.
            should do nothing if the result of the can_be_archived?() method is false.
        can_be_archived?() in the Book class
            should override the method from the parent class.
            should return true if parent's method returns true OR if cover_state equals to "bad".
            otherwise, it should return false.
        can_be_archived?() in the MusicAlbum class
            should override the method from the parent class.
            should return true if parent's method returns true AND if on_spotify equals true.
            otherwise, it should return false.
        can_be_archived?() in the Game class
            should override the method from the parent class.
            should return true if parent's method returns true AND if last_played_at is older than 2 years.
            otherwise, it should return false.
    Add unit tests for all implemented methods.
User interface

Create a main.rb file that will serve as your console app entry-point.
    Your console app, at the start, should:
        Present the user with a list of options to perform.
        Let users choose an option.
        If needed, ask for parameters for the option.
        Have a way to quit the app.
The following options should be available:
        List all books
        List all music albums
        List of games
        List all genres (e.g 'Comedy', 'Horror')
        List all labels (e.g. 'Gift', 'New')
        List all authors (e.g. 'Stephen King')
        Add a book
        Add a music album
        Add a game
All data should be preserved by saving collections in .json files.

Database

Created a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables

### CODE REVIEW

File: item.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  def add_source(source)
+    @sources << source
+  end
+
+  attr_writer :label
+
+  def add_author(author)
+    # @author << author
```
Comment (1GORDON) [inline item.rb:24]: Great job team 🤝

- [ ] Kindly remove all commented code from your project as this is regarded as a bad practice.

File: schema.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+CREATE TABLE genres (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    description TEXT NOT NULL
+);
+
+CREATE TABLE music_albums_genres (
+    music_album_id INT REFERENCES music_albums(id),
+    genre_id INT REFERENCES genres(id),
```
Comment (1GORDON) [inline schema.sql:60]: Great job team 🤝

- [ ] As stated in the project requirements, kindly ensure all foreign keys columns have corresponding indexes

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @techmoves, @AbdelazizSoliman, AND @AleWaweru ✋

Good job so far!

## Highlights!
- App runs as expected ✔️ 
- Data is preserved in the JSON files ✔️
- Methods are tested are tests are passing ✔️
- All aggregations (parent-child relationships) between are implemented✔️
- Each class from the UML diagram is created in a separate file ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] Great video presentation. However, you forgot to showcase a piece of code you are proud of building. Kindly record another video and please remember to showcase a piece of code you are proud of building. This is a project requirement for the video presentation.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions from the previous CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**